### PR TITLE
Internal sums of displayed categories

### PR DESCRIPTION
**Pull request by Daniel-Soe**
# Description

This PR defines internal sums of displayed categories. To make this definition I added the composition of displayed functors, displayed functors over a single base, displayed natural transformation and displayed adjoints. 

## Checklist

Before submitting a merge request, please check the items below:

- [x] The imports are sorted (use `find -type f -name \*.agda -or -name \*.lagda.md | xargs support/sort-imports.hs`)

- [x] All code blocks have "agda" as their language. This is so that
tools like Tokei can report accurate line counts for proofs vs. text.

- [x] Proofs are explained to a satisfactory degree; This is subjective,
of course, but proofs should be comprehensible to a hypothetical human
whose knowledge is limited to a working understanding of non-cubical
Agda, and the stuff your pages link to.

The following items are encouraged, but optional:

- [ ] If you feel comfortable, add yourself to the Authors page! Add a
profile picture that's recognisably "you" under support/pfps; The
picture should be recognisable at 128x128px, should look good in a
squircle, and shouldn't be more than 200KiB.

- [ ] If your contribution makes mention of a negative statement, but
does not prove the negative (perhaps because it would distract from the
main point), consider adding it to the counterexamples folder.

- [ ] If a proof can be done in both "cubical style", and "book HoTT
style", and you have the energy to do both, consider doing both!
However, it is **completely fine** to only do one! For instance, I
(Amélia) am much better at writing proofs "book-style".

If a commit affects many files without adding any content and you don't
want your name to appear on those pages (for example, treewide refactorings
or reformattings), start the commit message with `chore:` or include the word
`NOAUTHOR` anywhere.

### CODE REVIEW

**Comment by plt-amy** [comment]
@TOTBWF can you review the maths in this?

File: src/Cat/Displayed/NaturalTransformation.lagda.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    field
+      η′ : ∀ {x} (x′ : ℰ.Ob[ x ]) → ℱ.Hom[ η.η x ] (F′.₀′ x′) (G′.₀′ x′)
+      is-natural : ∀ {x y f} (x′ : ℰ.Ob[ x ]) (y′ : ℰ.Ob[ y ]) (f′ : ℰ.Hom[ f ] x′ y′)
+                    → PathP
+                       (λ i → ℱ.Hom[ η.is-natural x y f i ] (F′.₀′ x′) (G′.₀′ y′))
+                       (η′ y′ ℱ.∘′ F′.₁′ f′) (G′.₁′ f′ ℱ.∘′ η′ x′)
+```
+
+Let $\bf{F}$ and $\bf{G}$ be fibred functors. A displayed natural transformation
+$\bf{\eta} : \bf{F} \To \bf{G}$ is called **cartesian** if
```
Comment (TOTBWF) [inline src/Cat/Displayed/NaturalTransformation.lagda.md:70]: We should call these "vertical natural transformations" IMO.

File: src/Cat/Displayed/NaturalTransformation.lagda.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      module B = Precategory B
+      module ℰ = Displayed ℰ
+      module ℱ = Displayed ℱ
+      module F = Functor F
+      module F′ = Fibred-functor F′
+      module G′ = Fibred-functor G′
+      
+    field
+      η′ : ∀ {x} (x′ : ℰ.Ob[ x ]) → ℱ.Hom[ B.id ] (F′.₀′ x′) (G′.₀′ x′)
+      is-natural : ∀ {x y f} (x′ : ℰ.Ob[ x ]) (y′ : ℰ.Ob[ y ]) (f′ : ℰ.Hom[ f ] x′ y′)
```
Comment (TOTBWF) [inline src/Cat/Displayed/NaturalTransformation.lagda.md:90]:
```suggestion
      is-natural′ : ∀ {x y f} (x′ : ℰ.Ob[ x ]) (y′ : ℰ.Ob[ y ]) (f′ : ℰ.Hom[ f ] x′ y′)
```

File: src/Cat/Displayed/NaturalTransformation.lagda.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    private
+      module B = Precategory B
+      module ℰ = Displayed ℰ
+      module ℱ = Displayed ℱ
+      module F′ = Fibred-functor-single-base F′
+      module G′ = Fibred-functor-single-base G′
+      
+    field
+      η′ : ∀ {x} (x′ : ℰ.Ob[ x ]) → ℱ.Hom[ B.id ] (F′.₀′ x′) (G′.₀′ x′)
+      is-natural : ∀ {x y f} (x′ : ℰ.Ob[ x ]) (y′ : ℰ.Ob[ y ]) (f′ : ℰ.Hom[ f ] x′ y′)
```
Comment (TOTBWF) [inline src/Cat/Displayed/NaturalTransformation.lagda.md:121]:
```suggestion
      is-natural′ : ∀ {x y f} (x′ : ℰ.Ob[ x ]) (y′ : ℰ.Ob[ y ]) (f′ : ℰ.Hom[ f ] x′ y′)
```

**Review by TOTBWF — CHANGES_REQUESTED** [review]
First of all, thank you so much! I do have some minor naming/prose suggestions though, as well as one larger question about the math.

It is somewhat surprising to me that we are able to define internal sums even when `E` is not a fibration; the definition I'm familiar with requires left adjoints to reindexing + Beck Chevalley. Furthermore, I'm surprised you don't even need pullbacks in `B`. I do wonder if something goes wrong when `E` is a fibration yet `B` lacks pullbacks; I can't think of anything off the top of my head, but it's something to ponder.

**Comment by TOTBWF** [comment]
I actually need a lot of this code for some things I'm working on (Fibred Limits + Colimits); do you mind if I make some changes on this branch to get it merged quickly?

**Comment by Daniel-Soe** [comment]
@TOTBWF thank you for taking your time an reviewing my code. 

You are right, most literature defines internal sums only for a fibration $E$, but I never needed this assumption on $E$ so I dropped it.
For the case that $E$ is a fibration we need additionally that $B$ has pullbacks to prove that the codomain functor is a fibration which is needed to show that the family of $E$ `FamDisp` ist a fibration. 

I made all your requested changes except for the renaming, waiting for the response of @plt-amy. Now you can change whatever you want.

**Comment by TOTBWF** [comment]
Thanks so much!

P.S: When did github add $\LaTeX$!?

**Comment by Daniel-Soe** [comment]
> P.S: When did github add LATEX!?

I don't know, this was my first github comment and I just hoped that it would work😁

**Comment by TOTBWF** [comment]
Did some reorganizing and wrote some prose, but the code you gave was excellent! Thank you very much 🙂

**Review by TOTBWF — APPROVED** [review]

**Pull request merged**